tests/pkg/clients/destination_client.go

```diff
@@ -98,26 +102,58 @@ func NewDestinationClient(instanceConfig config.InstanceConfig, apiConfig Destin
 	}, nil
 }
 
-func (c *DestinationClient) CreateDestination(t *testing.T, destination Destination) {
+func (c *DestinationClient) CreateDestinationInDestService(t *testing.T, destination Destination, token string) {
```

Comment:
Because the c.httpClient rewrites the Authorization header. This leads to not having necessary data into the token. I can't really what was missing, but I think it was the subaccountId, serviceinstanceid, etc. The http client makes a call to the destination svc (ext-svc-mock) and the token gets validated and has its data extracted here: 

https://github.com/kyma-incubator/compass/blob/41b2354f571fa494cdbae4d2e09345d59715ec5b/components/external-services-mock/internal/destinationsvc/handler.go#L821

Reply:
Can you then:
- extract the creation of this new httpClient in a func and invoke it in the 4 places where such client is explicitly created 
- double check and delete the `c.httpClient` because to me it seems unused now